pkg/webui/console/views/device-general-settings/index.js

```diff
@@ -98,24 +123,137 @@ export default class DeviceGeneralSettings extends React.Component {
-    const { error } = this.state
+    const { device, isConfig, asConfig, jsConfig, nsConfig } = this.props
+
+    const isOTAA = isDeviceOTAA(device)
+    const { enabled: isEnabled } = isConfig
+    const { enabled: asEnabled } = asConfig
+    const { enabled: jsEnabled } = jsConfig
+    const { enabled: nsEnabled } = nsConfig
+
+    const isDisabled = !isEnabled
+    let isDescription = m.isDescription
+    if (isDisabled) {
+      isDescription = m.isDescriptionMissing
+    }
+
+    // 1. Disable the section if AS is not in cluster.
+    // 2. Disable the section if the device is OTAA, since no OTAA related fields are stored in the AS.
+    // 3. Disable the section if NS is not in cluster, since activation mode is unknown.
+    // 4. Disable the seciton if `application_server_address` is not equal to the cluster AS address
+    const sameAsAddress = getComponentBaseUrl(asConfig) === device.application_server_address
+    const asDisabled = !asEnabled || isOTAA || !nsEnabled || !sameAsAddress
+    let asDescription = m.asDescription
+    if (!asEnabled) {
+      asDescription = m.asDescriptionMissing
+    } else if (!nsEnabled) {
+      asDescription = m.activationModeUnknown
+    } else if (isOTAA) {
+      asDescription = m.asDescriptionOTAA
+    } else if (!sameAsAddress) {
+      asDescription = m.notInCluster
+    }
+
+    // 1. Disable the section if JS is not in cluster.
+    // 2. Disable the section if the device is ABP/Multicast, since JS does not store ABP/Multicast
+    // devices.
+    // 3. Disable the section if NS is not in cluster, since activation mode is unknown.
+    // 4. Disable the seciton if `join_server_address` is not equal to the cluster JS address
+    // 5. Disable the seciton if an external JS is used
+    const sameJsAddress = getComponentBaseUrl(jsConfig) === device.join_server_address
+    const externalJs = hasExternalJs(device)
+    const jsDisabled = !jsEnabled || !isOTAA || !nsEnabled || !sameJsAddress || externalJs
+    let jsDescription = m.jsDescription
+    if (!jsEnabled) {
+      jsDescription = m.jsDescriptionMissing
+    } else if (!nsEnabled) {
+      jsDescription = m.activationModeUnknown
+    } else if (!isOTAA) {
+      jsDescription = m.jsDescriptionOTAA
+    } else if (!sameJsAddress || externalJs) {
+      jsDescription = m.notInCluster
+    }
+
+    // 1. Disable the section if NS is not in cluster.
+    // 2. Disable the seciton if `network_server_address` is not equal to the cluster NS address
+    const sameNsAddress = getComponentBaseUrl(nsConfig) === device.network_server_address
+    const nsDisabled = !nsEnabled || !sameNsAddress
+    let nsDescription = m.nsDescription
+    if (!nsEnabled) {
+      nsDescription = m.nsDescriptionMissing
+    } else if (!sameNsAddress) {
+      nsDescription = m.notInCluster
+    }
 
     return (
       <Container>
         <IntlHelmet title={sharedMessages.generalSettings} />
         <Row>
           <Col lg={8} md={12}>
```

Comment:
```suggestion
          <Col lg={8} md={12} className={style.container}>
```
and then please add this:
```css
.container
  margin-top: $ls.s * -1